### DISCUSSION TITLE
Is there a way run Rector within another application?

### DISCUSSION THREAD

**Discussion by peterfox** [rectorphp/rector#7581] [Q&A] [answered]
I'd love to be able to use Rector to perform more upgrades in a staged approach. It would be awesome to build my own application to step through running different configurations quickly instead of having to manually tweak the config, run rector and see the results, then change the results again.

Is there a good entry point to running the rector process with an in-memory config programmatically?

**Comment by TomasVotruba** [comment, marked as answer]
There are 2 way to do this, based on your knowledge of hacking CLI apps written in Symfony.

1) You can duplicate the whole `ProcessCommand` - this requirest keeping your own "fork" up-to-date and will break eventually
2) You can use the `json` output formatter - easy and quick solution; I use this with the other PHP toolings to get metadata with `symfony/process` out of those
